wgengine/magicsock/magicsock_test.go

```diff
@@ -77,3 +78,58 @@ func TestDerpIPConstant(t *testing.T) {
 		t.Errorf("str %q != IP %v", derpMagicIPStr, derpMagicIP)
 	}
 }
+
+type stunStats struct {
+	mu       sync.Mutex
+	readIPv4 int
+	readIPv6 int
+}
+
+func serveSTUN(t *testing.T) net.Addr {
+	t.Helper()
+
+	// TODO(crawshaw): use stats to test re-STUN logic
+	var stats stunStats
+
+	pc, err := net.ListenPacket("udp4", ":3478")
+	if err != nil {
+		t.Fatalf("failed to open STUN listener: %v", err)
+	}
+	t.Cleanup(func() { pc.Close() })
```

Comment:
We officially switching to requiring Go 1.14 now? (Fine with me, but we need to update the CI first.)

Reply:
Will do later (too much going on for me to switch it now).